src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1167,17 +1140,196 @@ private boolean isSubtypeUncheckedInternal(Type t, Type s, boolean capture, Warn
                 return isSubtypeUncheckedInternal(t.getUpperBound(), s, false, warn);
             } else if (!s.isRaw()) {
                 Type t2 = asSuper(t, s.tsym);
-                if (t2 != null && t2.isRaw()) {
-                    if (isReifiable(s)) {
-                        warn.silentWarn(LintCategory.UNCHECKED);
+                if (t2 != null) {
+                    if (t2.isRaw()) {
+                        if (isReifiable(s)) {
+                            warn.silentWarn(LintCategory.UNCHECKED);
+                        } else {
+                            warn.warn(LintCategory.UNCHECKED);
+                        }
+                        return true;
                     } else {
-                        warn.warn(LintCategory.UNCHECKED);
+                        if (!isSameType(t2, s) &&
+                                t2.isReferenceProjection() == s.isReferenceProjection() &&
+                                structuralComparator.visit(t2, s)) {
+                            warn.warn(LintCategory.UNCHECKED);
+                            return true;
+                        }
+                        return false;
                     }
-                    return true;
                 }
             }
             return false;
         }
+        // where
+        StructuralTypeComparator structuralComparator = new StructuralTypeComparator();
+        class StructuralTypeComparator extends TypeRelation {
+            private Set<TypePair> cache = new HashSet<>();
+
+            @Override
+            public Boolean visitType(Type t, Type s) {
+                switch (t.getTag()) {
+                    case BYTE:
+                    case CHAR:
+                    case SHORT: case INT: case LONG:
+                    case FLOAT: case DOUBLE:
+                    case BOOLEAN: case VOID:
+                    case BOT: case NONE:
+                        return t.hasTag(s.getTag());
+                    case WILDCARD: //we shouldn't be here - avoids crash (see 7034495)
```

Comment:
I do not see a visitor method for wildcards, but I think you need one, as you might get to wildcards recursively, because of `visitClassType`

Reply:
we analyze the arguments of every parameterized type inside of `visitClassType`, so I think we should be covered for the wildcard case?

Reply:
I think we should be covered here as visitors: `StructuralTypeComparator` and `TypeArgStructComparator` are interdependent, so I think we should be fine there, unless I'm missing something